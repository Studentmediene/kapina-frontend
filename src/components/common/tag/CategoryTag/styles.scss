@import '../../../variables.scss';

.category {
  font-size: 16px;
  border-radius: 0.3em;
  margin-left: -0.5em;
  padding: 0.1em 0.5em;
  position: absolute;
}

@media screen and (max-width: $breakpoint-small) {
  .category {
    margin-top: 0 !important;
    font-size: 2px;
    border-radius: 0;
    margin-left: 0;
    position: relative;
  }
}