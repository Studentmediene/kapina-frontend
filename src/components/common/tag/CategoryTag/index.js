import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';

import styles from './styles.scss';

const CategoryTag = props => {
  const style = {
    backgroundColor: `${props.backgroundColor}`,
    color: `${props.textColor}`,
  };
  if (props.position === 'top') {
    style.marginTop = `${props.index * 2 + 0.4}em`;
  } else {
    style.marginTop = `-${(props.index + 1) * 2}em`;
  }

  return (
    <div className={styles.category} style={style}>
      {props.name}
    </div>
  );
};

CategoryTag.propTypes = {
  position: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  textColor: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
};

export default CategoryTag;
