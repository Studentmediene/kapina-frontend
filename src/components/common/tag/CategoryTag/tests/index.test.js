import React from 'react';
import { shallow } from 'enzyme';

import CategoryTag from '../';

describe('<CategoryTag />', () => {
  it('renders correctly on top', () => {
    const mockProps = {
      position: 'top',
      index: 0,
      name: 'categoryName',
      textColor: '#FFFFFF',
      backgroundColor: '#FFFFFF',
    };
    const tree = shallow(<CategoryTag {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly on bottom', () => {
    const mockProps = {
      position: 'bottom',
      index: 0,
      name: 'categoryName',
      textColor: '#FFFFFF',
      backgroundColor: '#FFFFFF',
    };
    const tree = shallow(<CategoryTag {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });
});
