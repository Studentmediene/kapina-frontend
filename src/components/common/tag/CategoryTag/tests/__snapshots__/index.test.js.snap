// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryTag /> renders correctly on bottom 1`] = `
<div
  className="category"
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "color": "#FFFFFF",
      "marginTop": "-2em",
    }
  }
>
  categoryName
</div>
`;

exports[`<CategoryTag /> renders correctly on top 1`] = `
<div
  className="category"
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "color": "#FFFFFF",
      "marginTop": "0.4em",
    }
  }
>
  categoryName
</div>
`;
